pkg/execution/queue/queue.go

```diff
@@ -0,0 +1,25 @@
+package queue
+
+import (
+	"context"
+	"time"
+)
+
+type Queue interface {
+	Enqueuer
+	Worker
+}
+
+type Enqueuer interface {
+	// Enqueue allows an item to be enqueued ton run at the given time.
+	Enqueue(context.Context, Item, time.Time) error
+}
+
+type Worker interface {
+	// Run is a blocking function which listens to the queue and executes the
+	// given function each time a new Item becomes available.
+	//
+	// This must only return an error if we fail to subscribe to the queue's
+	// implementation and can no longer process jobs.
+	Run(context.Context, func(context.Context, Item) error) error
```

Comment:
Would other implementations, e.g. RabbitMQ, extend this interface with functions like `Receive`/`Fetch` and `Poll` that passes the data to `Run`? I'm trying to imagine how one might write another implementation.